docs/content/guide/echo/typescript/snippets/subscription.ts

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright 2022 DXOS.org
+//
+
+import { Client } from '@dxos/client';
+
+import { Expando, create } from '@dxos/client/echo';
+
+const client = new Client();
+
+async () => {
+  await client.initialize();
+  if (!client.halo.identity.get()) await client.halo.createIdentity();
+
+  const space = await client.spaces.create();
+
+  const query = space.db.query({ type: 'task' });
+
+  const unsubscribeFn = query.subscribe(({ objects }) => {
+    objects.forEach(object => {
+      if (object.type === 'task') {
+        console.log('Do something with this task');
+      } else {
+        throw new Error('Non-task object returned (this will never happen)');
+      }
+    });
+  });
+
+  try {
```

Comment:
@yarolegovich: I used try/catch here instead of a [Context](https://github.com/dxos/dxos/blob/a7f191de8b3bd78fac9aa8c993c4080902a9eb2b/packages/common/context/src/context.ts#L40) because I wasn't confident enough in my understanding of Context to use it or document it.

Do you think this is good enough for now, or should we switch to `Context` in the code and explain how it works?